packages/web3/src/index.js

```diff
@@ -52,7 +52,7 @@ var Web3 = function Web3() {
 
     // overwrite package setProvider
     var setProvider = this.setProvider;
-    this.setProvider = function (provider, net) {
```

Comment:
line 56 `arguments` intakes `net`

Reply:
The `arguments` keyword contains all arguments passed, regardless of whether args are bound to variable names in the function signature.

A lot of best practices advice written today suggests that [the `arguments` keyword should be avoided](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments#Description). Perhaps this code would be more readable if it were rewritten to use a [rest parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters)?

Reply:
I would agree with this, I think I'll be tackling this issue next week (or this weekend). Theres unfortunately a lot of usages of `arguments` throughout the subpackages which inherintely use `net`. It got implemented like `super` unfortunately...